docs/configs-spec2006-tests/run_spec.py

```diff
@@ -0,0 +1,231 @@
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Jason Lowe-Power, Ayaz Akram, Hoa Nguyen
+
+""" Script to run a SPEC benchmark in full system mode with gem5.
+
+    Inputs:
+    * This script expects the following as arguments: name of the detailed
+    CPU, SPEC benchmark name, size of the benchmark, the path to the folder
+    to write SPEC log files, and paths to the kernel and the disk image.
+    * The SPEC benchmark name and the output folder must be provided, while 
+    the rest are optional.
+    * The default detailed CPU is KVM (not a detailed CPU).
+    * The default size is ref.
+    * The default path to the kernel is linux-stable/vmlinux-4.19.83
+    * The default path to the disk image is disk-image/spec2006/spec2006-image/spec2006
+
+    Outputs:
+    * TODO: simple performance statistics
+"""
+import os
+import sys
+import time
+
+import m5
+import m5.ticks
+from m5.objects import *
+
+import argparse
+
+from system import MySystem
+
+
+def writeBenchScript(dir, benchmark_name, size, output_path):
+    """
+    This method creates a script in dir which will be eventually
+    passed to the simulated system (to run a specific benchmark
+    at bootup).
+    """
+    input_file_name = '{}/run_{}_{}'.format(dir, benchmark_name, size)
+    with open(input_file_name,"w") as f:
+        f.write('{} {} {}'.format(benchmark_name, size, output_path))
+    return input_file_name
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="gem5 config file to run SPEC benchmarks")
+    parser.add_argument("-c", "--cpu", type = str,
+                        default = "kvm", help =" Name of the detailed CPU")
+    parser.add_argument("-b", "--benchmark", type = str, required = True,
```

Comment:
If this is required it shouldn't be an optional argument, it should be positional. See https://docs.python.org/3/library/argparse.html#name-or-flags

Reply:
I like required arguments with explicit parameter names in command as it makes the command clearer and easier to understand. We don't need to remember the order the positional parameters. All that, I don't see it very popular in practice, so I'll switch back to positional arguments.